contracts/mainnet/connectors/sushi-incentive/main.sol

```diff
@@ -34,17 +35,19 @@ abstract contract SushipswapIncentiveResolver is Helpers, Events {
         token1 = changeEthAddrToWethAddr(token1);
         token2 = changeEthAddrToWethAddr(token2);
         amount = getUint(getId, amount);
-        (uint256 _pid, uint256 _version, address lpTokenAddr) = _getPoolId(
-            token1,
-            token2
-        );
-        setUint(setId, _pid);
-        require(_pid != uint256(-1), "pool-does-not-exist");
-        TokenInterface lpToken = TokenInterface(lpTokenAddr);
+        if(data.poolId == uint256(-1)|| data.version <= 0 || data.lpToken == address(0)){
+            data = _getPoolId(
+                token1,
+                token2
+            );
+        }
+        setUint(setId, data.poolId);
+        require(data.poolId != uint256(-1), "pool-does-not-exist");
+        TokenInterface lpToken = TokenInterface(data.lpToken);
         lpToken.approve(address(masterChef), amount);
-        _deposit(_pid, amount, _version);
+        _deposit(data, amount);
         _eventName = "LogDeposit(uint256,uint256,uint256)";
-        _eventParam = abi.encode(_pid, _version, amount);
+        _eventParam = abi.encode(data.poolId, data.version, amount);
```

Comment:
@pradyuman-verma Let's emit tokenA and tokenB too?

Reply:
okay